euphonic/spectra/base.py

```diff
@@ -1,28 +1,21 @@
+"""Classes for spectral data"""
 # pylint: disable=no-member
 
 from abc import ABC, abstractmethod
-import collections
 import copy
-from functools import partial, reduce
-from itertools import product, repeat, starmap
+from functools import partial
 
 import math
-import json
 from numbers import Integral, Real
-from operator import contains
-from typing import (Any, Callable, Dict, Generator, List, Literal, Optional,
+from typing import (Any, Callable, Dict, List, Literal, Optional,
```

Comment:
Given Euphonic is now 3.10+, use modern typing. I.e. `list`, `dict`, `|`-unions, import `Callable` from `collections.abc`, etc.

Reply:
Yeah, I had this stuff in mind to do as well. Separate PR seems harmless, there's enough in this one for now.